charts/kafka-operator/templates/operator-deployment-with-webhook.yaml

```diff
@@ -212,7 +212,7 @@ spec:
           {{- end }}
           ports:
           {{- if .Values.webhook.enabled }}
-            - containerPort: {{ .Values.webhook.serverPort | default 443 }}
+            - containerPort: {{ .Values.webhook.serverPort | default 9443 }}
```

Comment:
Can you provide some reference for the reasoning behind this change?

Reply:
It looks like openshift, by default, due to its security constraints, it does not allow binding on ports lower than 1024.

Reply:
Thanks for the info! 

Do you guys think this change will cause any backward compatibility issues for the existing users when they upgrade the Koperator?

Reply:
the compatibility would be an issue because of the service definition using numerical values for targetPort rather than container port name references.  Added a comment on the service template diff.  

k8s doc ref:  https://kubernetes.io/docs/concepts/services-networking/service/#defining-a-service
> Port definitions in Pods have names, and you can reference these names in the targetPort attribute of a Service. ...
> This works even if there is a mixture of Pods in the Service using a single configured name, with the same network protocol available via different port numbers. This offers a lot of flexibility for deploying and evolving your Services. For example, you can change the port numbers that Pods expose in the next version of your backend software, without breaking clients.

Reply:
->Darren: I think the webhook port where the webhook is available is mainly used by K8s API server to send objects to, especially in the case of the container port itself (CMIIW). The service port is a bit more exposed, I will address that a bit further below.
A user - technically speaking - could have built on top of the default behavior (default 443 port), but we can signal the change of that in the release notes.
It's arguable whether we should classify this as major (breaking) change. I'm suggesting to stay on the safe side and chalk it up as a breaking (default behavior) change in the notes and give it an extra highlight, but otherwise we shouldn't be too concerned, it is easily fixable by code building on top of this - and hopefully people didn't hard code it but derived the dynamic value from the manifest.
In addition we are still on 0.x exactly because of these things, the API can change including open port numbers.

I will address Tim's comment on a later chain where it also popped up.

Reply:
to clarify, this is only a breaking change because new upgraded endpoints with different containerPort numerical values won't automatically have their ports mapped when under the non-upgraded service.  Essentially if the service definition is changed to use the port name then the k8s service port to containerPort mapping works for all cases.  So for upgrade, compatibility is preserved as long as the k8s service is upgraded concurrently with or before the koperator endpoint using the new port number for the webhook port--assuming both new and old versions of koperator deployments used the same port NAME.